substrate/primitives/arithmetic/src/fixed_point.rs

```diff
@@ -1033,6 +1088,15 @@ macro_rules! implement_fixed {
 				let _ = AsMaxEncodedLen { _data: $name::min_value() };
 			}
 
+			#[test]
+			fn u32_roundtrip_matches() {
+				let fixed = $name::from_u32(u32::MAX);
+				assert_eq!(fixed.0, (u32::MAX as $inner_type) * $name::DIV);
+
+				let val = fixed.to_u32();
+				assert_eq!(val, u32::MAX);
```

Comment:
Can you add some more cases here please? Maybe also a random generated value in a loop.

We have some fuzz tests in `sp-arithmetic-fuzzer`, probably overkill to add it there, just saying.